caffe2/operators/gather_ranges_to_dense_op.h

```diff
@@ -45,14 +45,19 @@ class GatherRangesToDenseOp final : public Operator<Context> {
     // Initialize the empty and mismatch counter.
     for (const auto i : c10::irange(OutputSize())) {
       (void)i; // Suppress unused variable warning
+      totalRanges_.push_back(0);
       emptyRanges_.push_back(0);
       mismatchedRanges_.push_back(0);
       mismatchedLengths_.push_back(set<int>());
     }
   }
 
   ~GatherRangesToDenseOp() noexcept override {
-    if (totalRanges_ > minObservation_) {
+    bool exceedMinObservation = false;
+    for (int i = 0; i < totalRanges_.size(); i++) {
+      exceedMinObservation |= totalRanges_[i] > minObservation_;
+    }
```

Comment:
Use range loop if possible

```suggestion
    for (const auto range : totalRanges_) {
      exceedMinObservation |= range > minObservation_;
    }
```

Reply:
will update. thanks